src/main/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -157,40 +157,45 @@ void nextContainer() throws IllegalArgumentException, CRAMException {
 
         parser.getRecords(container, cramRecords, validationStringency);
 
-        if (container.sequenceId == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
-            refs = new byte[]{};
-            prevSeqId = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
-        } else if (container.sequenceId == Slice.MULTI_REFERENCE) {
-            refs = null;
-            prevSeqId = Slice.MULTI_REFERENCE;
-        } else if (prevSeqId < 0 || prevSeqId != container.sequenceId) {
-            final SAMSequenceRecord sequence = cramHeader.getSamFileHeader()
-                    .getSequence(container.sequenceId);
-            refs = referenceSource.getReferenceBases(sequence, true);
-            if (refs == null) {
-                throw new CRAMException(String.format("Contig %s not found in the reference file.", sequence.getSequenceName()));
-            }
-            prevSeqId = container.sequenceId;
+        switch (container.getSliceType()) {
+            case UNMAPPED:
+                referenceBases = new byte[]{};
+                prevSeqId = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
+                break;
+            case MULTI_REFERENCE:
+                referenceBases = null;
+                prevSeqId = Slice.MULTI_REFERENCE;
+                break;
+            default:
+                if (prevSeqId != container.getSequenceId()) {
```

Comment:
removed the `prevSeqId < 0` check: since we know that the slice type is single-ref, we also know that the current sequence ID here is 0+.  So it will differ from any negative values.